### ISSUE TITLE
Lintrule "invalid-void" should permit `Promise<void>` 

### ISSUE THREAD

**Issue by Akuukis** [palantir/tslint#4834]
### Bug Report

- __TSLint version__: 5.19.0
- __TypeScript version__: 3.5.3
- __Running TSLint via__: CLI

I generally support `invalid-void` lint rule
> Restrict invalid usage of void type. For example, as property type or in type enumerations. I believe the void type should be used as return type only and be alone (without any kind of intersection types - see examples below).

But I suggest to permit `Promise<void>` return type also. IMO it's wrong to force to change all of them to `Promise<undefined>`, that only escalates to TS error: `A function whose declared type is neither 'void' nor 'any' must return a value.ts(2355)`

#### TypeScript code being linted

```ts
async function myAsyncFn(): Promise<void> {
    return Promise.resolve()
}
```

#### Actual behavior
Fails lint rule `invalid-void` #4732

#### Expected behavior
Passes

**Comment by adidahiya** [comment]
duplicate of #4828
